Lib/zipimport.py

```diff
@@ -556,12 +556,20 @@ def _read_directory(archive):
         finally:
             fp.seek(start_offset)
     _bootstrap._verbose_message('zipimport: found {} names in {!r}', count, archive)
+
+    # Add implicit directories.
     for name in list(files):
-        while (i := name.rstrip(path_sep).rfind(path_sep)) >= 0:
+        while True:
+            i = name.rstrip(path_sep).rfind(path_sep)
+            if i < 0:
+                break
             name = name[:i + 1]
             if name in files:
                 break
             files[name] = None
+            count += 1
+    _bootstrap._verbose_message('zipimport: added {} implicit directories in {!r}',
+                                count, archive)
```

Comment:
This `count` has not be recounted.

Reply:
Confirmed - `count` should probably be reset to 0 before the loop... or maybe a different variable should be used.

Reply:
Oh, thank you for your catch. Will be fixed by #121375.